spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonClientConfiguration.java

```diff
@@ -155,6 +166,28 @@ public ServerIntrospector serverIntrospector() {
 		return new DefaultServerIntrospector();
 	}
 
+	@PostConstruct
+	public void preprocess() {
+		setProp(name, DeploymentContextBasedVipAddresses.key(), name);		
```

Comment:
Yeah, It is definitely redundant...I dont know enough about the internals of how Eureka is used to create a ribbon client dynamically vs. having a ribbon client explicitly defined with RibbonClientConfiguration.